From f9804d46d01971b5924137d85e5e189d6f96b895 Mon Sep 17 00:00:00 2001
From: zaclimon <isaacpateau05@gmail.com>
Date: Sat, 10 Sep 2016 20:21:17 -0400
Subject: [PATCH] LegacyCameraDevice: Remove preview resolution check

Some devices that have been able to record in a certain resolution
before (Eg. Nexus 4 in 1080p) aren't able now because of this check.

I know that removing the check isn't the preferred solution but
it can be seen as a way for the moment.
---
 .../hardware/camera2/legacy/LegacyCameraDevice.java       | 15 +--------------
 1 file changed, 1 insertion(+), 14 deletions(-)

diff --git a/core/java/android/hardware/camera2/legacy/LegacyCameraDevice.java b/core/java/android/hardware/camera2/legacy/LegacyCameraDevice.java
index 3e79118..8793bef 100644
--- a/core/java/android/hardware/camera2/legacy/LegacyCameraDevice.java
+++ b/core/java/android/hardware/camera2/legacy/LegacyCameraDevice.java
@@ -361,20 +361,7 @@ public class LegacyCameraDevice implements AutoCloseable {
                         }
                     }
 
-                    if (!ArrayUtils.contains(sizes, s)) {
-                        if (flexibleConsumer && (s = findClosestSize(s, sizes)) != null) {
-                            sizedSurfaces.add(new Pair<>(output, s));
-                        } else {
-                            String reason = (sizes == null) ? "format is invalid." :
-                                    ("size not in valid set: " + Arrays.toString(sizes));
-                            Log.e(TAG, String.format("Surface with size (w=%d, h=%d) and format " +
-                                    "0x%x is not valid, %s", s.getWidth(), s.getHeight(),
-                                    surfaceType, reason));
-                            return BAD_VALUE;
-                        }
-                    } else {
-                        sizedSurfaces.add(new Pair<>(output, s));
-                    }
+                    sizedSurfaces.add(new Pair<>(output, s));
                     // Lock down the size before configuration
                     setSurfaceDimens(output, s.getWidth(), s.getHeight());
                 } catch (BufferQueueAbandonedException e) {
-- 
2.7.4
